### PR TITLE
API error handling

### PR DESCRIPTION
**Pull request by floriandejonckheere**
:warning: ~~Review and merge #94 first.~~

- Drop TODOs in every API saga
- Add API module to keep track of API request statuses
- Add `ApiDimmer` that displays a spinner on a pending request
- Move signin saga to new error handling flow

Asynchronous API errors are now handled with actions.

### CODE REVIEW

**Comment by Dromin** [comment]
@floriandejonckheere What is the reason that api status is kept in a separate module, instead of having each module keep its own api status? I expect that components in each module will need to access their own api request status, for example to display a spinner while fetching. Or am I misunderstanding something?

File: app/modules/api/tests/reducer.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      },
+    };
+
+    const nextState = {
+      foobar: {
+        status: statusTypes.SUCCESS,
+      },
+    };
+
+    expect(reducer(prevState, setStatusAction)).toEqual(nextState);
```
Comment (Dromin) [inline app/modules/api/tests/reducer.test.js]: In reducer tests, please also test that the reducer does not accidentally mutate the state:
```
const resultState = reducer(prevState, setStatusAction);
expect(resultState).toEqual(next.state);
expect(resultState).not.toBe(prevState);
[...*]
```
\* If resultState contains objects that had their values changed, also test if these are not equal by reference. See `modules/content-items/tests/reducer.test.js` for examples.

**Comment by Dromin** [comment]
Ok, makes sense.

File: app/modules/authentication/saga/api/confirmation.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ export const apiPostConfirmationSaga = function* (
     yield call(ConfirmationApi.post, email);
   }
   catch (error) {
-    // TODO
+    // TODO: fix saga error handling
+    throw error;
```
Comment (Dromin) [inline app/modules/authentication/saga/api/confirmation.js]: What is the plan here? Dispatch an error action?

File: app/modules/api/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+/* Action types */
+
+// Reducer actions
+export type SetStatusAction = {
+  type: typeof SET_STATUS,
+  payload: {
+    request: string,
+    status: StatusType,
+    error?: Error,
```
[the remark this answers is not in the document]
Comment (floriandejonckheere) [reply app/modules/api/actionTypes.js]: This is the ErrorAction, which used to be the way to return errors in the action creator. This has been replaced with `throw new Error` now, see #101.

File: app/modules/authentication/saga/api/confirmation.js (hunk 2 above)
Comment (floriandejonckheere) [reply app/modules/authentication/saga/api/confirmation.js]: This is not included in this branch yet, but that's the idea. If a request was successful, dispatch an API request success action, if not an API request failure action.

File: app/modules/api/actionTypes.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import type { Error } from 'types/error';
+
+import type { StatusType } from './model';
+
+/* Action constants */
+
+// Reducer actions
+export const SET_STATUS_IN_STATE: 'api/SET_STATUS_IN_STATE_IN_STATE' = 'api/SET_STATUS_IN_STATE_IN_STATE';
```
Comment (JasperDhaene) [inline app/modules/api/actionTypes.js]: SET_STATUS_IN_STATE**_IN_STATE** is probably a replace error?

Comment (floriandejonckheere) [reply app/modules/api/actionTypes.js]: Oops! Y u approve tho

**Pull request merged**